### DISCUSSION TITLE
Use Windows 10 accent colour in OMP theme

### DISCUSSION THREAD

**Discussion by SiJiL82** [JanDeDobbeleer/oh-my-posh#2302] [Ideas]
Is there a way we can use our Windows 10 accent colour in an OMP theme?
This one:
![image](https://user-images.githubusercontent.com/65114782/169646795-0ddee9a1-d9ea-4279-acd1-0fe6113acbc0.png)
Windows Terminal uses it for progress bars:
![image](https://user-images.githubusercontent.com/65114782/169646824-80fd1969-a35d-4809-bcdb-78081b87a0fd.png)
It seems there are ways to get it in code, not sure if any of them are doable with how OMP works though?
https://devblogs.microsoft.com/oldnewthing/20170405-00/?p=95905

**Comment by JanDeDobbeleer** [comment]
Not yet but it seems entirely possible!

**Reply by JanDeDobbeleer** [reply]
https://user-images.githubusercontent.com/2492783/169666064-29d8a65f-5259-4a43-8811-9da26bf2ffdf.mov

Needs some cleanup, but look at that.

**Reply by SiJiL82** [reply]
That's awesome!
Does it work with the "Automatically pick an accent colour from my background" too?
If you set your background to auto change every X minutes, the accent colour changes at the same time - does this pick up the new colour straight away?

**Reply by JanDeDobbeleer** [reply]
It should yes.

**Comment by JanDeDobbeleer** [comment]
This is now available by using the `accent` color keyword on Windows. Use `accent_color` in the [root config](https://ohmyposh.dev/docs/configuration/overview#general-settings) as a fallback color when on another OS.

**Comment by SiJiL82** [comment]
Just grabbed the update from Winget, it works great!
I have a massive folder of wallpapers that I have set on a 5m slideshow, so it's fun to see my prompt match the wallpaper general colour each time I use it :D
My work laptop uses a branded accent colour to match the company logo, so having my prompt match that nicely will be pretty good too!

**Comment by SiJiL82** [comment]
Are there 2 different colours available when you get the Windows accent colour? Just noticed when I saw the Winget progress bar, that my OMP accent colour seems to be the "background" colour - the bit where the progress bar isn't yet filled:
![image](https://user-images.githubusercontent.com/65114782/170819964-54af56ef-3c8f-4b36-ae06-cb34e4a75948.png)
The "foreground" colour on the progress bar matches with what the accent colour looks like in Windows settings:
![image](https://user-images.githubusercontent.com/65114782/170819983-6cffc9b6-faa6-43cd-a8d6-5ada0cd610df.png)

**Reply by JanDeDobbeleer** [reply]
@SiJiL82 I only validated this on 11 where it seems to work just fine? There's an alpha later available that I don't really use, but that's the color I get yes.

**Reply by JanDeDobbeleer** [reply]
@SiJiL82 seems like there's something annoying about that win32 call. BUT, I found a [post](https://stackoverflow.com/questions/3560890/vista-7-how-to-get-glass-color) that explains that and everything else. Seems to to be correct now (colors match when I color pick them). See #2347

**Reply by SiJiL82** [reply]
It seems to be related to the "Automatically pick an accent colour from my background" option.
With it enabled, my accent colour is this light blue, but OMP gets the darker variant:
![image](https://user-images.githubusercontent.com/65114782/170860339-c92c7a0f-c4e5-4552-8cf9-8a97ffc60bfb.png)
However if I manually pick the same colour (from the "recent colours" list), OMP matches it:
![image](https://user-images.githubusercontent.com/65114782/170860367-3096d849-bd6d-492e-85b9-6d2fcb13d656.png)

**Reply by JanDeDobbeleer** [reply]
@SiJiL82 did you update to the latest version?

**Reply by SiJiL82** [reply]
I have 7.93.1 (the latest on Winget)

**Reply by JanDeDobbeleer** [reply]
@SiJiL82 in that case it's a windows bug (as the post clearly indicates to go to that value and it is correct when using the color picker).

**Reply by SiJiL82** [reply]
That's a shame! It's ok most of the time, just occasionally you get a really dark colour that makes it impossible to see the text on it :D

**Reply by SiJiL82** [reply]
This post suggests there is indeed an issue with the registry entry storing the "wrong" value when you have "Automatically pick an accent colour from my background" selected.
https://stackoverflow.com/questions/50840395/c-sharp-console-get-windows-10-accent-color
There's a comment saying `you can use a 'couple' of calls, and it works when "Automatically pick an accent color from my background" is enabled as well`
Not sure if that code they've provided is usable?

**Comment by SagnikPradhan** [comment]
Quite weird, but it doesn't work at all for me, even with manually choosing the accent color. I am changing the background color value to accent here.
![normal](https://user-images.githubusercontent.com/26499576/192684170-3d8f310c-9726-490b-b373-3e54c87f9b9c.png)
![bug](https://user-images.githubusercontent.com/26499576/192684192-901c3a56-be1d-415f-b2d0-430ec140f6c0.png)

**Reply by JanDeDobbeleer** [reply]
@SagnikPradhan can you show me the config? Seems like there's an issue with the template/color here as it's not correctly parsed.

**Reply by SagnikPradhan** [reply]
<details>
<summary>
Yes, here's the configuration. I really appreciate the quick response ❤️ 
</summary>

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",

  "blocks": [
    {
      "alignment": "left",
      "segments": [
        {
          "type": "path",
          "style": "plain",
          "foreground": "black",
          "properties": {
            "style": "agnoster_full",
            "mapped_locations": {
              "D:\\Projects": ""
            }
          }
        },
        {
          "type": "time",
          "powerline_symbol": "\ue0b0",
          "foreground": "#fff",
          "background": "accent",
          "style": "powerline"
        },
        {
          "type": "git",
          "style": "plain",
          "foreground": "lightRed",
          "template": " <#DDB15F>git(</>{{ .HEAD }}<#DDB15F>)</>",
          "properties": {
            "branch_icon": ""
          }
        },
        {
          "foreground": "#DCB977",
          "style": "plain",
          "template": " \uf119",
          "type": "exit"
        }
      ],
      "type": "prompt"
    }
  ],

  "console_title_template": "{{.Shell}} {{.Folder}}",

  "final_space": true,

  "version": 2
}
```

</details>

**Reply by JanDeDobbeleer** [reply]
@SagnikPradhan on which OS are you trying this? This happens when you don't have for example `"accent_color": "#DCB977",` set in the config on another platform (or if the accent color can't be resolved). This works:

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "segments": [
        {
          "type": "path",
          "style": "plain",
          "foreground": "black",
          "properties": {
            "style": "agnoster_full",
            "mapped_locations": {
              "D:\\Projects": ""
            }
          }
        },
        {
          "type": "time",
          "powerline_symbol": "\ue0b0",
          "foreground": "#fff",
          "background": "accent",
          "style": "powerline"
        },
        {
          "type": "git",
          "style": "plain",
          "foreground": "lightRed",
          "template": " <#DDB15F>git(</>{{ .HEAD }}<#DDB15F>)</>",
          "properties": {
            "branch_icon": ""
          }
        },
        {
          "foreground": "#DCB977",
          "style": "plain",
          "template": " \uf119",
          "type": "exit"
        }
      ],
      "type": "prompt"
    }
  ],
  "console_title_template": "{{.Shell}} {{.Folder}}",
  "final_space": true,
  "accent_color": "#DCB977",
  "version": 2
}
```

Can you share the output of `oh-my-posh debug`?

**Reply by SagnikPradhan** [reply]
I am running this on windows 10, also that doesn't seem to work for me either. [Here](https://hastebin.com/ulofabapog.apache)'s the output of `oh-my-posh debug`.

**Reply by SagnikPradhan** [reply]
Oh my god, I just realized it wasn't updating even after running the commands from the guide. Ended up uninstalling it completely and using the Microsoft store version.

**Reply by JanDeDobbeleer** [reply]
Indeed as you had a very old version. Could be useful to see why it wasn't updating.
